src/util.js

```diff
@@ -276,6 +281,85 @@ function copyDescriptors(from, to, filter) {
     }
 }
 
+global.eventsToHandlers = global.eventsToHandlers ? global.eventsToHandlers :
+    {};
+// This function should not be used for methods that return a value
+// Using this function implies small performance overhead, should be used only
+// for debug methods
+function hookTo(prototypeObject, methodName, extension) {
+    var existingMethodImpl = prototypeObject[methodName];
+    var extensions = null;
+    if (typeof existingMethodImpl === "function") {
+        extensions = global.eventsToHandlers[methodName];
+        if (!extensions) {
+            extensions = global.eventsToHandlers[methodName]
+                = [];
+        }
+        if (extensions.indexOf(extension) >= 0) {
+            throw new Error("Trying to register same extension method twice");
+        }
+        extensions.push(extension);
+        prototypeObject[methodName] = generateFunctionFromExtensions(
+            extensions, methodName);
+
+    } else if (typeof existingMethodImpl === "undefined") {
+        if (typeof extension === "function") {
+            extensions = [extension];
+            prototypeObject[methodName] = generateFunctionFromExtensions(
+                extensions, methodName);
+        } else {
+            extensions = [];
+            prototypeObject[methodName] =
+                generateFunctionFromExtensions(extensions, methodName);
+        }
+    } else {
+        throw new Error("Trying to wrap " + typeof existingMethodImpl +
+            ", expecting a function or undefined");
+    }
+}
+
+function unhookFrom(prototypeObject, methodName, extension) {
+    var extensions = global.eventsToHandlers[methodName];
+    if (!Array.isArray(extensions)) {
+        throw new Error("Trying to unhook from method, that was not extended");
+    }
+    var index = extensions.indexOf(extension);
+    if (index < 0) {
+        throw new Error("Trying to unhook extension function " + extension +
+            " that was not hooked to current prototype, extensions that were" +
+            " hooked are: " + extensions);
+    }
+    extensions.splice(index, 1);
+    prototypeObject[methodName] = generateFunctionFromExtensions(extensions);
+}
+
+function generateFunctionFromExtensions(extensions, methodName) {
```

Comment:
This method looks kind of needlessly complicated and not very performant. Any reason why we do all this wrapping?